src/filesystem.ts

```diff
@@ -0,0 +1,57 @@
+import localforage from 'localforage'
+
+import FileSystem from './fs'
+import * as dataRoot from './data-root'
+import { FS_CID, FS_TIMESTAMP , authenticatedUsername } from './common'
+
+
+/**
+ * Load a user's file system.
+ *
+ * @param username Optional, username of the user to load the file system from.
+ *                 Will try to load the file system of the authenticated user
+ *                 by default. Throws an error if there's no authenticated user.
+ */
+export async function loadFileSystem(username?: string): Promise<FileSystem> {
+  let cid, fs
+
+  // Determine username
+  username = username || (await authenticatedUsername() || undefined)
+  if (!username) throw new Error("User hasn't authenticated yet")
+
+  const lastLocalChange = await localforage.getItem(FS_TIMESTAMP) as number || 0
+  const currentTime = Date.now()
+
+  // If our last file-system change was over 15 minutes ago
+  if (currentTime - lastLocalChange > 30 * 60 * 1000) {
+    cid = await dataRoot.lookup(username)
```

Comment:
This is done to avoid data loss, because either the DNS doesn't update fast enough, or the system/browser caches the DNS too long.

This could most likely be done better, so if anyone has any quick fixes for this, let me know.